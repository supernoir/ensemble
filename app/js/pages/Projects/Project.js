import React from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';
import intl from 'react-intl-universal';
import { Container, Breadcrumb, Segment, Header, Table, Divider, Button, Icon } from 'semantic-ui-react';

export default class Project extends React.Component {
	constructor(){
		super();
		this.state = {
			project: [],
			cast   : [],
		};
		this.getCharacter = this.getCharacter.bind(this);
	}

	componentDidMount(){
		axios.get(`http://localhost:3030/project/${this.props.match.params.id}`)
			.then(res => {
				if (res.data.project.cast !== void 0) {
					let rawCast = res.data.project.cast.split(',');
					rawCast.map(member => this.getCharacter(member));
				}
				this.setState({ project: res.data.project });
			}
			);
	}

	getCharacter(id) {
		return axios.get(`http://localhost:3030/character/${id}`)
			.then(res => {
				this.setState({
					cast: [
						...this.state.cast,
						res.data.character
					]
				});
			}
			);
	}

	render(){
		return(
			<Container>
				<Breadcrumb>
					<Breadcrumb.Section link>
						<Link to="/">{intl.get('component.dashboard')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section link>
						<Link to="/projects">{intl.get('entity.projects')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section active>
						{this.state.project.title}
					</Breadcrumb.Section>
				</Breadcrumb>

				<Segment>
					<Header as='h2'>
						{this.state.project.title}
						<Header.Subheader>{this.state.project.series}</Header.Subheader>
					</Header>
					<Divider/>

					<p>
						<Icon name='info circle'/>{this.state.project.desc}
					</p>

					<Divider/>


					<Table celled striped>
						<Table.Header>
							<Table.Row>
								<Table.HeaderCell colSpan='3'>{intl.get('entity.characters')}</Table.HeaderCell>
							</Table.Row>
						</Table.Header>

						<Table.Body>
							{this.state.cast !== void 0 || this.state.cast.length > 0
								? this.state.cast.map(member => {
									return <Table.Row>
										<Table.Cell>
											<Link to={`/character/${member._id}`}>{member.first_name} {member.last_name}</Link>
										</Table.Cell>
									</Table.Row>;
								})
								: null
							}
						</Table.Body>
					</Table>
					<Divider/>
					<Button>
						{intl.get('project.action-edit')}
					</Button>
					<Button>
						{intl.get('project.action-delete')}
					</Button>
				</Segment>

			</Container>
		);
	}
}