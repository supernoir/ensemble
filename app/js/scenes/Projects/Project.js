import React from 'react';
import { Link } from 'react-router-dom';
import intl from 'react-intl-universal';
import { Container, Breadcrumb, Segment, Header, Table, Divider, Button, Icon, Label } from 'semantic-ui-react';
import Loader from '../../basics/Loader';

export default class Project extends React.Component {
	componentDidMount() {
		this.props.getProjectById(this.props.match.params.id);
	}

	render() {
		return this.props.loading
			? <Loader loading={this.props.loading} />
			: <Container>
				<Breadcrumb>
					<Breadcrumb.Section link>
						<Link to="/">{intl.get('component.dashboard')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section link>
						<Link to="/projects">{intl.get('entity.projects')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section active>
						{this.props.project.title}
					</Breadcrumb.Section>
				</Breadcrumb>

				<Segment>
					<Header as="h2">
						{this.props.project.title}
						<Header.Subheader>{this.props.project.series}</Header.Subheader>
					</Header>
					<Divider />

					<p>
						<Icon name="info circle" />{this.props.project.desc}
					</p>

					{
						this.props.project.tags !== void 0
							? <Label.Group tag>
								{
									this.props.project.tags.map((tag,index) => {
										return <Label key={`${tag}-${index}`}>{tag}</Label>;
									})
								}
							</Label.Group>
							: null
					}
					<Divider />

					<Table celled striped>
						<Table.Header>
							<Table.Row>
								<Table.HeaderCell colSpan="3">{intl.get('entity.characters')}</Table.HeaderCell>
							</Table.Row>
						</Table.Header>
						<Table.Body>

						</Table.Body>
					</Table>
					<Divider />
					<Button>
						{intl.get('project.action-edit')}
					</Button>
					<Button>
						{intl.get('project.action-delete')}
					</Button>
				</Segment>

			</Container>;
	}
}
