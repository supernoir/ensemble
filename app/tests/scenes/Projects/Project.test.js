import React from 'react';
import { shallow } from 'enzyme';

// Import Mocks
import mockProject from '../../mocks/mockProject';

// Import Components
import Project from '../../../js/scenes/Projects/Project';

test('Project -- Snapshot Test', () => {
	const wrapper = shallow(
		<Project
			loading={false}
			match={{ params: 1 }}
			getProjectById={id => jest.fn(id)}
			project={mockProject}
			deleteSpecificProject={id => {
				jest.fn(id);
			}}
		/>
	);
	expect(wrapper).toMatchSnapshot();
});
