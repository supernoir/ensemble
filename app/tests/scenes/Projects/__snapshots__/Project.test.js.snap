// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project -- Snapshot Test 1`] = `
<Container>
  <Breadcrumb>
    <BreadcrumbSection>
      <Link
        replace={false}
        to="/"
      />
    </BreadcrumbSection>
    <BreadcrumbDivider />
    <BreadcrumbSection>
      <Link
        replace={false}
        to="/projects"
      />
    </BreadcrumbSection>
    <BreadcrumbDivider />
    <BreadcrumbSection
      active={true}
    >
      The Passersby
    </BreadcrumbSection>
  </Breadcrumb>
  <SegmentGroup>
    <Segment>
      <Header
        as="h2"
      >
        The Passersby
      </Header>
    </Segment>
    <SegmentGroup
      horizontal={true}
    >
      <Segment>
        <Header
          as="h4"
        />
        <Label
          color="blue"
          content=""
          icon="thumbtack"
          size="large"
        />
      </Segment>
      <Segment>
        <Header
          as="h4"
        />
        <Label
          content="The Passersby"
          icon="paperclip"
          size="large"
        />
      </Segment>
    </SegmentGroup>
    <Segment>
      <Header
        as="h4"
      />
      <p>
        Dublin story
      </p>
    </Segment>
    <Segment>
      <Header
        as="h4"
      />
      <Label
        key="Drama-0"
      >
        Drama
      </Label>
    </Segment>
    <Segment>
      <Header
        as="h4"
      />
      <Table
        as="table"
        celled={true}
        striped={true}
      >
        <TableHeader
          as="thead"
        >
          <TableRow
            as="tr"
            cellAs="td"
          >
            <TableHeaderCell
              as="th"
            />
            <TableHeaderCell
              as="th"
            >
              I18N Circles
            </TableHeaderCell>
          </TableRow>
        </TableHeader>
        <TableBody
          as="tbody"
        >
          <TableRow
            as="tr"
            cellAs="td"
            key="David-0"
          >
            <TableCell
              as="td"
            >
              David
            </TableCell>
            <TableCell
              as="td"
            >
              -
            </TableCell>
          </TableRow>
          <TableRow
            as="tr"
            cellAs="td"
            key="Sean-1"
          >
            <TableCell
              as="td"
            >
              Sean
            </TableCell>
            <TableCell
              as="td"
            >
              -
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </Segment>
    <Segment>
      <Header
        as="h4"
      />
      <LabelGroup
        tag={true}
      >
        <Label
          key="example-0"
        >
          example
        </Label>
        <Label
          key="sample-1"
        >
          sample
        </Label>
        <Label
          key="test-2"
        >
          test
        </Label>
      </LabelGroup>
    </Segment>
    <Segment>
      <Link
        replace={false}
        to="/editproject/5d13a6864ea603c5dcff98d8"
      >
        <Button
          as="button"
        />
      </Link>
      <Button
        as="button"
        content=""
        onClick={[Function]}
      />
    </Segment>
  </SegmentGroup>
</Container>
`;
